RestoreBase.php

```diff
@@ -152,12 +152,16 @@ public function importAstDB($data) {
 			foreach($content as $family => $extra) {
 				if(!is_array($extra)){// content is not array eg:/DAYNIGHT/C      : NIGHT
 					$this->log(sprintf(_('Importing AstDB family %s/%s  : %s'),$key,$family,$extra));
-					$this->FreePBX->astman->database_put($key,$family,$extra);
+					if($this->FreePBX->astman->connected()){
+						$this->FreePBX->astman->database_put($key,$family,$extra);
```

Comment:
This code should fail but throw a meaningful exception. If you proceed without restoring the astdb entries, things will likely fail in a weird way that is hard to troubleshoot and reproduce. Therefore, I recommend changing it to throw an exception if not connected, rather than proceeding as usual or throwing whatever it currently throws.